central/complianceoperator/v2/report/manager/utils/results_aggregator_test.go

```diff
@@ -0,0 +1,547 @@
+		Description:  "description",
+		Status:       status,
+		Rationale:    "rationale",
+		Instructions: "instructions",
+	}
+}
+
+func assertStatus(t *testing.T, expected storage.ComplianceOperatorCheckResultV2_CheckStatus, actual *checkStatus) {
+	switch expected {
+	case storage.ComplianceOperatorCheckResultV2_PASS:
+		assert.Equal(t, 1, actual.totalPass)
+		assert.Equal(t, 0, actual.totalFail)
+		assert.Equal(t, 0, actual.totalMixed)
+	case storage.ComplianceOperatorCheckResultV2_FAIL:
+		assert.Equal(t, 0, actual.totalPass)
+		assert.Equal(t, 1, actual.totalFail)
+		assert.Equal(t, 0, actual.totalMixed)
+	default:
+		assert.Equal(t, 0, actual.totalPass)
+		assert.Equal(t, 0, actual.totalFail)
+		assert.Equal(t, 1, actual.totalMixed)
+	}
+}
+
+func assertResult(t *testing.T, tcase walkByQueryTestCase, row *report.ResultRow) {
+	assert.Equal(t, tcase.check.GetClusterName(), row.ClusterName)
+	assert.Equal(t, tcase.check.GetCheckName(), row.CheckName)
+	assert.Equal(t, tcase.check.GetDescription(), row.Description)
+	assert.Equal(t, tcase.check.GetStatus().String(), row.Status)
+	assert.Equal(t, tcase.check.GetRationale(), row.Rationale)
+	assert.Equal(t, tcase.check.GetInstructions(), row.Instructions)
+	if tcase.expectedProfiles != nil {
+		expProfiles, _ := tcase.expectedProfiles()
+		if len(expProfiles) < 1 {
+			assert.Equal(t, DATA_NOT_AVAILABLE, row.Profile)
+		} else {
+			require.Len(t, expProfiles, 1)
+			assert.Equal(t, fmt.Sprintf("%s %s", expProfiles[0].GetName(), expProfiles[0].GetProfileVersion()), row.Profile)
+		}
+	}
+	if tcase.expectedRemediations != nil {
+		expRemediations, _ := tcase.expectedRemediations()
+		if len(expRemediations) == 0 {
+			assert.Equal(t, NO_REMEDIATION, row.Remediation)
+		} else {
+			expRemediationNames := make([]string, 0, len(remediations))
+			for _, remediation := range expRemediations {
+				expRemediationNames = append(expRemediationNames, remediation.GetName())
+			}
+			assert.Equal(t, strings.Join(expRemediationNames, ","), row.Remediation)
+		}
+	}
+	if tcase.expectedRules != nil {
+		expRules, _ := tcase.expectedRules()
+		if len(expRules) != 1 {
+			assert.Equal(t, DATA_NOT_AVAILABLE, row.ControlRef)
+		} else {
+			if tcase.expectedBenchmarks == nil {
+				assert.Equal(t, DATA_NOT_AVAILABLE, row.ControlRef)
+			} else {
+				expBench, _ := tcase.expectedBenchmarks()
+				if len(expBench) == 0 {
+					assert.Equal(t, DATA_NOT_AVAILABLE, row.ControlRef)
+				} else {
+					if tcase.expectedControls != nil {
+						expControls, _ := tcase.expectedControls()
+						if len(expControls) == 0 {
+							assert.Equal(t, DATA_NOT_AVAILABLE, row.ControlRef)
+						} else {
+							expControlInfos := make([]string, 0, len(expControls))
+							for _, c := range expControls {
+								expControlInfos = append(expControlInfos, fmt.Sprintf("%s %s", c.Standard, c.Control))
+							}
+							assert.Equal(t, strings.Join(expControlInfos, ","), row.ControlRef)
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
Wow, so much nesting. Any chance to flatten it maybe?

Reply:
Done 🙂